src/validators/api/endpoints.py

```diff
@@ -0,0 +1,124 @@
+
+
+async def get_validators(request: Request) -> Response:
+    deposit_data = request.app.state.deposit_data
+
+    validators = await get_available_validators_for_registration(
+        keystore=None, deposit_data=deposit_data, run_check_deposit_data_root=False
+    )
+    validators = [v for v in validators if v.public_key not in pending_validator_registrations]
+
+    # get next validator index for exit signature
+    latest_public_keys = await get_latest_network_validator_public_keys()
+    next_validator_index = NetworkValidatorCrud().get_next_validator_index(list(latest_public_keys))
+
+    return JSONResponse(
+        [
+            {'public_key': validator.public_key, 'index': index}
+            for index, validator in enumerate(validators, next_validator_index)
+        ]
+    )
+
+
+async def approve_validators(request: Request) -> Response:
+    if settings.validators_registration_mode != ValidatorsRegistrationMode.API:
+        return JSONResponse(
+            {'error': 'validators registration mode must be "API"'},
+            status_code=HTTP_400_BAD_REQUEST,
+        )
+
+    try:
+        payload = await request.json()
+    except json.JSONDecodeError:
+        return JSONResponse({'error': 'invalid json'}, status_code=HTTP_400_BAD_REQUEST)
+
+    adapter = TypeAdapter(list[ValidatorApproval])
+    try:
+        approvals: list[ValidatorApproval] = adapter.validate_python(payload)
+    except ValidationError as e:
+        return JSONResponse({'error': json.loads(e.json())}, status_code=HTTP_400_BAD_REQUEST)
+
+    if not approvals:
+        return JSONResponse({'error': 'invalid validators'}, status_code=HTTP_400_BAD_REQUEST)
+
+    deposit_data = request.app.state.deposit_data
+
+    validators = await get_available_validators_for_registration(
+        keystore=None, deposit_data=deposit_data
+    )
+
+    error = await _validate_approvals(approvals, validators)
+    if error is not None:
+        return JSONResponse({'error': error}, status_code=HTTP_400_BAD_REQUEST)
+
+    # There may be lag between GET and POST requests.
+    # During this time new assets may be staked into the vault.
+    # In this case validators list will be longer than approvals list.
+    # Filter validators by requested public keys.
+    validators = validators[: len(approvals)]
+
+    for validator, approval in zip(validators, approvals):
+        validator.exit_signature = approval.exit_signature
+
+    pending_validator_registrations.extend([v.public_key for v in validators])
+    asyncio.create_task(
+        register_and_remove_pending_validators(
+            keystore=None, deposit_data=deposit_data, validators=validators
+        )
+    )
+
+    return JSONResponse({})
+
+
+async def _validate_approvals(
+    approvals: list[ValidatorApproval], validators: list[Validator]
+) -> Any:
+    """
+    Business logic validation
+    :return: error
+    """
+    if len(validators) < len(approvals):
```

Comment:
I think we should be able to proceed with such case as well. If there are more approvals that available validators for the registration, we can take `approvals[:len(validators)]`.